sqe-database-access/TextRepository.cs

```diff
@@ -424,13 +906,36 @@ UserInfo                editionUser
 					// }
 					//
 					// previousSignData = newSignData;
-					newSigns.Add(newSignData);
+
+					// Collect the info for each new sign interpretation
+					newlyCreatedSigns.Add(newSignData);
+
+					// Added by Ingo: We have to set the anchorsBefore to the NextSignItnterpretationsId of the last
+					// created Sign.
+					updatedSignInterpretationIds.AddRange(anchorsBefore);
+					anchorsBefore.Clear();
+
+					anchorsBefore.AddRange(
+							newSignData.SignInterpretations.Select(
+									si => si.SignInterpretationId.GetValueOrDefault()));
 				}
 
 				transactionScope.Complete();
 			}
 
-			return newSigns;
+			// Rebuild any affected materialized sign streams
+			// Do not await it, we don't care if it finishes before this
+			// request is over.
+			// TODO: change RequestMaterializationAsync to take an array,
+			// it should be able to check how many sign streams would need a rebuild.
+			if (materializeSignStream && updatedSignInterpretationIds.Any())
+			{
+				_materializationRepository.RequestMaterializationAsync(
```

Comment:
I am not sure if we really want to leave something running after the request is done - in this process. This may be OK for the short term, but really if this is a lengthy operation, we should probably off-load it to a different process.

Reply:
Agreed, what are your suggestions for doing that?  Do we use BackgroundService?